subprojects/docs/src/docs/userguide/directory_layout.adoc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[directory_layout]]
+= Directory Layout
+
+[.lead]
+Gradle uses two main directories to perform and manage its work: the <<#directory_layout:gradle_user_home_directory>> and the <<#directory_layout:project_root_directory>>.
+
+
+[[directory_layout:gradle_user_home_directory]]
+== Gradle user home directory
+
+The Gradle user home directory has roughly the following structure:
+
+[listing]
+----
+.
+├── caches // <1>
+│   ├── 4.8 // <2>
+│   ├── 4.9 // <2>
```

Comment:
I don't think Asciidoctor automatically aligns annotation markers to the right. Could you please manually right-align them?

Reply:
why should they be right aligned? I think that would make them harder to read.

Reply:
:o: What I typically do here is try to align these callouts vertically in the left-most column. That said, I don't think we have any data to suggest this is better. 

I'm very happy that you use callouts at all. We finally can do this now that we're all asciidoc :)